Add Pulumi to integrations-extras

isaac-pulumi opened the pull request:
### What does this PR do?

Initial PR to create the Pulumi tile

### Motivation

User/Community awareness: a new integration with Pulumi is available.

### Review checklist

- [X] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [X] Feature or bugfix has tests
- [X] Git history is clean
- [X] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

ChristineTChen reviewed (changes requested):
Hey @isaac-pulumi ,

did you create this via the [ddev](https://datadoghq.dev/integrations-core/setup/#stable) cli tool? It looks like you're missing the `/image` directory and `assets/service_checks.json` file

```
ddev create --type tile Pulumi
```

Also please include the integration to the [CODEOWNERS](https://github.com/DataDog/integrations-extras/blob/master/.github/CODEOWNERS) file

ChristineTChen commented on .github/CODEOWNERS:
```diff
@@ -67,6 +67,7 @@
 /portworx/                               @pault84 paul@portworx.com
 /postman/                                @shashankawasthi88  integrations-partnerships@postman.com
 /pulsar/                                 @zzzming itestmycode@gmail.com
+/pulumi/                                 team@pulumi.com
```
Please include a github username as well

isaac-pulumi replied:
```suggestion
/pulumi/                                 @isaac-pulumi team@pulumi.com
```

sarina-dd commented on pulumi/README.md:
```diff
@@ -0,0 +1,64 @@
+## Overview
+
+[Pulumi][1] is a modern infrastructure as code platform that enables cloud engineering teams to define, deploy and manage cloud resources on any cloud using their favorite programming languages.
+The Datadog provider for Pulumi can be used to provision any of the cloud resources available in Datadog. The Datadog provider must be configured with credentials to deploy and update resources in Datadog.
```
```suggestion
The Pulumi integration is used to provision any of the cloud resources available in Datadog. This integration must be configured with credentials to deploy and update resources in Datadog.
```

sarina-dd commented on pulumi/README.md:
```diff
@@ -0,0 +1,64 @@
+
+### Installation
+
+The [Pulumi Datadog provider][2] uses the Datadog SDK to manage and provision resources.
```
```suggestion
The [Pulumi Datadog integration][2] uses the Datadog SDK to manage and provision resources.
```

sarina-dd commented on pulumi/README.md:
```diff
@@ -0,0 +1,64 @@
+
+### Configuration
+
+1. [Sign-up for a free or commercial Pulumi account][3]
```
```suggestion
1. [Sign up for a free or commercial Pulumi account][3]
```

sarina-dd commented on pulumi/README.md:
```diff
@@ -0,0 +1,64 @@
+
+2. [Install Pulumi][4]
+
+3. Once obtained, there are two ways to communicate your Datadog authorization tokens to Pulumi:
```
```suggestion
3. Once obtained, there are two ways to set your Datadog authorization tokens for Pulumi:
```

sarina-dd commented on pulumi/README.md:
```diff
@@ -0,0 +1,64 @@
+3. Once obtained, there are two ways to communicate your Datadog authorization tokens to Pulumi:
+
+
+- Set the environment variables `DATADOG_API_KEY` and `DATADOG_APP_KEY`:
```
```suggestion
Set the environment variables `DATADOG_API_KEY` and `DATADOG_APP_KEY`:
```

sarina-dd commented on pulumi/README.md:
````diff
@@ -0,0 +1,64 @@
+export DATADOG_API_KEY=XXXXXXXXXXXXXX && export DATADOG_APP_KEY=YYYYYYYYYYYYYY
+```
+
+- Or, set them using configuration if you prefer that they be stored alongside your Pulumi stack for easy multi-user access:
````
```suggestion
Or, set them using configuration if you prefer that they be stored alongside your Pulumi stack for easier multi-user access:
```

sarina-dd commented on pulumi/README.md:
````diff
@@ -0,0 +1,64 @@
+pulumi config set datadog:apiKey XXXXXXXXXXXXXX --secret && pulumi config set datadog:appKey YYYYYYYYYYYYYY --secret
+```
+
+Remember to pass `--secret` when setting `datadog:apiKey` and `datadog:appKey` so that they are properly encrypted.
````
```suggestion
**Note**: Pass `--secret` when setting `datadog:apiKey` and `datadog:appKey` so that they are properly encrypted.
```

sarina-dd commented on pulumi/README.md:
```diff
@@ -0,0 +1,64 @@
+
+Remember to pass `--secret` when setting `datadog:apiKey` and `datadog:appKey` so that they are properly encrypted.
+
+4. Run `pulumi new` to initialize a project directory for your infrastructure stack and follow the [API documentation][5] to define new Metrics, Monitors, Dashboards or other resources.
```
```suggestion
4. Run `pulumi new` to initialize a project directory for your infrastructure stack and follow the [API documentation][5] to define new metrics, monitors, dashboards, or other resources.
```

sarina-dd reviewed (commented):
Docs review completed

isaac-pulumi commented on .github/CODEOWNERS:
```diff
@@ -286,6 +287,9 @@
 /pulsar/*metadata.csv                    @zzzming itestmycode@gmail.com @DataDog/documentation
 /pulsar/manifest.json                    @zzzming itestmycode@gmail.com @DataDog/documentation
 /pulsar/README.md                        @zzzming itestmycode@gmail.com @DataDog/documentation
+/pulumi/*metadata.csv                    team@pulumi.com @DataDog/documentation
```
```suggestion
/pulumi/*metadata.csv                    @isaac-pulumi team@pulumi.com @DataDog/documentation
```

isaac-pulumi commented on .github/CODEOWNERS:
```diff
@@ -286,6 +287,9 @@
 /pulsar/manifest.json                    @zzzming itestmycode@gmail.com @DataDog/documentation
 /pulsar/README.md                        @zzzming itestmycode@gmail.com @DataDog/documentation
+/pulumi/*metadata.csv                    team@pulumi.com @DataDog/documentation
+/pulumi/manifest.json                    team@pulumi.com @DataDog/documentation
```
```suggestion
/pulumi/manifest.json                    @isaac-pulumi team@pulumi.com @DataDog/documentation
```

isaac-pulumi commented on .github/CODEOWNERS:
```diff
@@ -286,6 +287,9 @@
 /pulsar/README.md                        @zzzming itestmycode@gmail.com @DataDog/documentation
+/pulumi/*metadata.csv                    team@pulumi.com @DataDog/documentation
+/pulumi/manifest.json                    team@pulumi.com @DataDog/documentation
+/pulumi/README.md                        team@pulumi.com @DataDog/documentation
```
```suggestion
/pulumi/README.md                        @isaac-pulumi team@pulumi.com @DataDog/documentation
```

isaac-pulumi reviewed (commented):
Addressed feedback from @sarina-dd and @ChristineTChen

isaac-pulumi commented:
> Hey @isaac-pulumi ,
> 
> did you create this via the [ddev](https://datadoghq.dev/integrations-core/setup/#stable) cli tool? It looks like you're missing the `/image` directory and `assets/service_checks.json` file
> 
> ```
> ddev create --type tile Pulumi
> ```
> 
> Also please include the integration to the [CODEOWNERS](https://github.com/DataDog/integrations-extras/blob/master/.github/CODEOWNERS) file

@ChristineTChen I think I addressed the issues you call out here. I don't have any images for README.md - are those required and do you have examples?

ChristineTChen commented on pulumi/manifest.json:
```diff
@@ -0,0 +1,37 @@
+  "creates_events": false,
+  "short_description": "Infrastructure as code for any cloud using your favorite programming languages",
+  "guid": "58481132-c79e-4659-8064-7cdaabbbc981",
+  "support": "core",
```
```suggestion
  "support": "contrib",
```

isaac-pulumi replied:
@ChristineTChen I think I've addressed this -- please let me know if there are other changes needed.

ChristineTChen reviewed (changes requested):
@isaac-pulumi

It looks like you're still missing `service_checks.json` (even if there is no service check because this is tile-only, it's part of the intg packaging. Here's an [example](https://github.com/DataDog/integrations-extras/blob/master/federatorai/assets/service_checks.json), it belongs in `pulumi/assets/`: 

Logos belong in `pulumi/assets/logos`, here's an example of the [logos/ directory](https://github.com/DataDog/integrations-extras/tree/master/federatorai/assets/logos)

FlorianVeaux commented:
Hey @isaac-pulumi 
@ChristineTChen's comment is still relevant, tests are failing at the moment. Can you look into re-adding a service-checks.json file?

isaac-pulumi reviewed (commented):
Addressed comments from @ChristineTChen

isaac-pulumi reviewed (commented):
addressing changes from @ChristineTChen

isaac-pulumi commented:
> Hey @isaac-pulumi
> @ChristineTChen's comment is still relevant, tests are failing at the moment. Can you look into re-adding a service-checks.json file?

@FlorianVeaux I think I've addressed the changes requested above - please confirm.

FlorianVeaux commented on pulumi/manifest.json:
```diff
@@ -0,0 +1,37 @@
+    "windows"
+  ],
+  "public_title": "PULUMI",
+  "categories": [
+    "cat-orchestration",
+    "cat-aws",
+    "cat-azure",
+    "cat-cloud",
+    "cat-google-cloud",
+    "cat-configuration-deployment"
```
The categories here must be from the existing ones you can find here: https://docs.datadoghq.com/integrations/

isaac-pulumi replied:
@FlorianVeaux I believe that I've addressed this (the docs are not very clear but  I've standardized on category names per your feedback).

FlorianVeaux reviewed (commented):
Thanks for addressing!
I have requested our design team to create appropriate images for our app based on the svg you provided. We'll need that before we can deploy it.
I'm also trying the integration into a development environment and I'll make some screenshots to show you what it looks like once we have the logos.
Lastly, while syncing I realize that the categories in the manifest.json don't match with what we use in our documentation. Can you have a look at that?

FlorianVeaux commented on pulumi/manifest.json:
```diff
@@ -0,0 +1,37 @@
+    "mac_os",
+    "windows"
+  ],
+  "public_title": "PULUMI",
```
```suggestion
  "public_title": "Datadog-Pulumi Integration",
```

FlorianVeaux commented:
![image](https://user-images.githubusercontent.com/22912273/134697746-7ec00392-67f0-4cb8-b499-b663617743af.png)

Looks like that at the moment, image missing but that's expected for now

isaac-pulumi commented:
> ![image](https://user-images.githubusercontent.com/22912273/134697746-7ec00392-67f0-4cb8-b499-b663617743af.png)
> 
> Looks like that at the moment, image missing but that's expected for now

I've addressed the other issues you've called out @FlorianVeaux and fixed the "all caps" issue for the public_title field.

fanny-jiang commented on pulumi/manifest.json:
```diff
@@ -0,0 +1,37 @@
+    "azure",
+    "cloud",
+    "google cloud",
+    "configuration deployment"
```
small nit

```suggestion
    "configuration & deployment"
```

isaac-pulumi reviewed (commented):
@fanny-jiang I believe I've addressed your feedback here.

fanny-jiang commented:
Hi @isaac-pulumi, one more thing
 
> @ChristineTChen I think I addressed the issues you call out here. I don't have any images for README.md - are those required and do you have examples?

The `/images` directory can be left empty with an empty `.gitkeep` file, for example: https://github.com/DataDog/integrations-extras/tree/85fd2c284f6137550ade5546839f331e15d84ee0/calico/images
